### PR TITLE
added Additional information section + tests

### PR DESCRIPTION
**Pull request by vladimir32pa**

### CODE REVIEW

File: app/section/sections/browser-additional-information/ui.test.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      .reply(200, {});
+  }
+};
+
+const pageSetup = async (t, withMarketingData = false) => {
+  mocks(withMarketingData);
+  await t.navigateTo('http://localhost:1234/S100000-001/section/browser-additional-information');
+};
+
+fixture.only('Show Solution Description page')
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: remove the .only pretty please

File: app/section/sections/browser-additional-information/ui.test.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+  const mainAdvice = Selector('[data-test-id="section-main-advice"]');
+
+  await t
+    .expect(mainAdvice.innerText).eql('Answer the following questions about your browser based solution');
+});
+
+test('should render all the advice of the section', async (t) => {
+  await pageSetup(t);
+
+  const sectionManifest = new ManifestProvider().getSectionManifest('solution-description');
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: don't think you are using this here. Could you remove this please.

File: app/section/sections/browser-additional-information/ui.test.js [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+
+  const additionalAdvice = Selector('[data-test-id="section-additional-advice"]');
+
+  await t
+    .expect(additionalAdvice.innerText).eql('This information is diplayed on the Marketing page for buyers to view');
+});
+
+test('should render the solution summary question', async (t) => {
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: Change name of the test

File: app/section/sections/browser-additional-information/ui.test.js [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+  await pageSetup(t);
+
+  const summaryQuestion = Selector('[data-test-id="question-additional-information"]');
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: change name of const

File: app/section/sections/browser-additional-information/ui.test.js [hunk 5]
```diff
@@ -0,0 +1,181 @@
...
+  await pageSetup(t);
+
+  nock('http://localhost:8080')
+    .put('/api/v1/Solutions/S100000-001/sections/browser-additional-information')
+    .reply(400, {
+      maxLength: ['additional-information'],
+    });
+
+  const oneHundredCharacters = '0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789';
+  const thousandCharacters = oneHundredCharacters.repeat(10);
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: actually I don't think we need this. The mock will return the error regardless of what we put in that text-area

File: app/section/sections/browser-additional-information/ui.test.js [hunk 6]
```diff
@@ -0,0 +1,181 @@
...
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const additionalInformation = Selector('[data-test-id="question-additional-information"]');
+  
+  const submitButton = Selector('[data-test-id="section-submit-button"] button');
+  
+  await t
+  .expect(errorSummary.exists).notOk()
+  .typeText(additionalInformation, `${thousandCharacters}0`, { paste: true })
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: Don't need this anymore

File: app/section/sections/browser-additional-information/ui.test.js [hunk 7]
```diff
@@ -0,0 +1,181 @@
...
+  await pageSetup(t);
+
+  nock('http://localhost:8080')
+    .put('/api/v1/Solutions/S100000-001/sections/browser-additional-information')
+    .reply(400, {
+      maxLength: ['additional-information'],
+    });
+
+  const oneHundredCharacters = '0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789';
+  const thousandCharacters = oneHundredCharacters.repeat(10);
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: Don't need this

File: app/section/sections/browser-additional-information/ui.test.js [hunk 8]
```diff
@@ -0,0 +1,181 @@
...
+
+  const getLocation = ClientFunction(() => document.location.href);
+
+  await t
+    .expect(errorSummary.exists).notOk()
+    .typeText(additionalInformation, `${thousandCharacters}0`, { paste: true })
+    .click(submitButton)
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li:nth-child(1) a').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#additional-information')
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: I would just test the anchor behavior. The above is already covered in the test above. Just keep the `click(submitButton)`

File: app/section/sections/browser-additional-information/ui.test.js [hunk 9]
```diff
@@ -0,0 +1,181 @@
...
+    .click(errorSummaryList.find('li:nth-child(1) a'))
+    .expect(getLocation()).contains('/S100000-001/section/browser-additional-information#additional-information');
+});
+
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: I would add another test that makes sure it goes back to the brower-based dashboard on a successful submit.

we've done that for the other sections that are browser-based subSections. Something like below ->
```
test('should goto the browser based dashboard when clicking the submit button', async (t) => {
  await pageSetup(t, 200, connectivityAndResolutionMarketingData);

  nock('http://localhost:8080')
    .put('/api/v1/Solutions/S100000-001/sections/connectivity-and-resolution')
    .reply(200, connectivityAndResolutionMarketingData);

  nock('http://localhost:8080')
    .get('/api/v1/Solutions/S100000-001/sections/browser-based')
    .reply(200, {});

  const submitButton = Selector('[data-test-id="section-submit-button"] button');
  const getLocation = ClientFunction(() => document.location.href);

  await t
    .expect(submitButton.exists).ok()
    .click(submitButton)
    .expect(getLocation()).contains('/S100000-001/dashboard/browser-based');
});
```

File: app/section/sections/browser-additional-information/manifest.json [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+  "title": "Browser based - additional information",
+  "mainAdvice": "Answer the following questions about your browser based solution",
+  "additionalAdvice": [
+    "This information is diplayed on the Marketing page for buyers to view"
+  ],
+  "questions": {
+    "additional-information": {
+      "mainAdvice": "Do you have any additional browser based information (optional)",
+      "additionalAdvice": "Add any additional information or requirement your Solution needs to function",
+      "footerAdvice": "You have 500 characters remaining",
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/manifest.json]: Sorry this isn't right. Can you change to the following `"You can enter up to 500 characters"` - I think the prototype was out of date

File: app/section/sections/browser-additional-information/ui.test.js [hunk 11]
```diff
@@ -0,0 +1,181 @@
...
+  await t
+  .expect(errorSummary.exists).notOk()
+  .typeText(additionalInformation, `${thousandCharacters}0`, { paste: true })
+  .click(submitButton)
+  .expect(errorSummary.exists).ok()
+  .expect(errorSummaryList.find('li').count).eql(1)
+  .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql('Additional information is over the character limit')
+  .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#additional-information')
+  .expect(additionalInformation.find('.nhsuk-textarea--error').exists).ok()
+  .expect(additionalInformation.find('.nhsuk-error-message').innerText).eql('Error:\nAdditional information is over the character limit')
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: can you indent all of these please. I'm not sure if your vscode has the eslint plugin. But you should be seeing red squigle lines to highlight this

File: app/section/sections/browser-additional-information/ui.test.js [hunk 12]
```diff
@@ -0,0 +1,181 @@
...
+
+  const oneHundredCharacters = '0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789';
+  const thousandCharacters = oneHundredCharacters.repeat(10);
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const additionalInformation = Selector('[data-test-id="question-additional-information"]');
+  
+  const submitButton = Selector('[data-test-id="section-submit-button"] button');
+
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: get trailing spaces eslint error here

File: app/section/sections/browser-additional-information/ui.test.js [hunk 13]
```diff
@@ -0,0 +1,181 @@
...
+      maxLength: ['additional-information'],
+    });
+
+  const oneHundredCharacters = '0123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789';
+  const thousandCharacters = oneHundredCharacters.repeat(10);
+
+  const errorSummary = Selector('[data-test-id="error-summary"]');
+  const errorSummaryList = Selector('.nhsuk-error-summary__list');
+  const additionalInformation = Selector('[data-test-id="question-additional-information"]');
+
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: getting trailing spaces eslint error here

File: app/section/sections/browser-additional-information/ui.test.js [hunk 14]
```diff
@@ -0,0 +1,181 @@
...
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: getting trailing spaces eslint error here

File: app/section/sections/browser-additional-information/ui.test.js [hunk 15]
```diff
@@ -0,0 +1,181 @@
+import nock from 'nock';
+import { Selector, ClientFunction } from 'testcafe';
+import { ManifestProvider } from '../../../manifestProvider';
+import dashboardWithCompleteSections from '../../../../fixtures/dashboardWithCompleteSections.json';
+
+const additionalInformationData = {
+  'additional-information': 'The solution additional information'
```
Comment (alibombaye) [inline app/section/sections/browser-additional-information/ui.test.js]: getting missing trailing comma eslint error here

File: app/section/sections/browser-additional-information/ui.test.js (hunk 1 above)
Comment (alibombaye) [reply app/section/sections/browser-additional-information/ui.test.js]: Also name of fixture too

File: app/section/sections/browser-additional-information/manifest.json [hunk 16]
```diff
@@ -0,0 +1,21 @@
...
+      "mainAdvice": "Do you have any additional browser based information (optional)",
+      "additionalAdvice": "Add any additional information or requirement your Solution needs to function",
+      "footerAdvice": "You can enter up to 500 characters",
+      "type": "textarea-field",
+      "errorResponse": {
+        "maxLength": "Additional information is over the character limit"
+      }
+    }
+  },
+  "submitText" : "Save and return to browser based",
```
Comment (sam-roberts-bjss) [inline app/section/sections/browser-additional-information/manifest.json]: Observation: Some of the content will be addressed at a later date. All the other screens state "Save and return".

File: app/section/sections/browser-additional-information/ui.test.js [hunk 17]
```diff
@@ -0,0 +1,186 @@
+import nock from 'nock';
+import { Selector, ClientFunction } from 'testcafe';
+import { ManifestProvider } from '../../../manifestProvider';
```
Comment (sam-roberts-bjss) [inline app/section/sections/browser-additional-information/ui.test.js]: ManifestProvider is not used?

File: app/section/sections/browser-additional-information/ui.test.js [hunk 18]
```diff
@@ -0,0 +1,186 @@
...
+    .expect(additionalInformation.find('[data-test-id="textarea-field-footer"]').innerText).eql('You can enter up to 500 characters');
+});
+
+test('should populate the text fields with existing data', async (t) => {
+  pageSetup(t, true);
+
+  const theQuestions = Selector('form');
+
+  await t
+    .expect(theQuestions.find('[data-test-id="question-additional-information"]').find('textarea').value).eql('The solution additional information')
```
Comment (sam-roberts-bjss) [inline app/section/sections/browser-additional-information/ui.test.js]: Missing ';'

File: app/section/sections/browser-additional-information/ui.test.js (hunk 13 above)
Comment (sam-roberts-bjss) [reply app/section/sections/browser-additional-information/ui.test.js]: Still getting this error?

File: app/section/sections/browser-additional-information/ui.test.js [hunk 19]
```diff
@@ -0,0 +1,186 @@
...
+
+  await t
+    .expect(errorSummary.exists).notOk()
+    .click(submitButton)
+    .expect(errorSummary.exists).ok()
+    .expect(errorSummaryList.find('li').count).eql(1)
+    .expect(errorSummaryList.find('li:nth-child(1)').innerText).eql('Additional information is over the character limit')
+    .expect(errorSummaryList.find('li:nth-child(1) a').getAttribute('href')).eql('#additional-information')
+    .expect(additionalInformation.find('.nhsuk-textarea--error').exists).ok()
+    .expect(additionalInformation.find('.nhsuk-error-message').innerText).eql('Error:\nAdditional information is over the character limit')
```
Comment (sam-roberts-bjss) [inline app/section/sections/browser-additional-information/ui.test.js]: Missing ';'

File: testcafeRunner.js [hunk 20]
```diff
@@ -16,7 +16,7 @@ createTestcafe('localhost')
 
     return tc.createRunner()
       .src(['**/*ui.test.js'])
-      .browsers('chrome:headless')
+      .browsers('chrome')
```
Comment (sam-roberts-bjss) [inline testcafeRunner.js]: Check your merge, I'm pretty sure Fawad has changed this file

Comment (vladimir32pa) [reply testcafeRunner.js]: reverted back to `.browsers('chrome:headless')`, git should handle the rest

**Pull request merged**